pkg/segment/results/blockresults/runningstats.go

```diff
@@ -598,3 +607,59 @@ func (rr *RunningBucketResults) GetRunningStatsBucketValues() ([]utils.CValueEnc
 	}
 	return retVal, rr.count
 }
+
+func (rs runningStats) GetRunningStatJSON() RunningStatsJSON {
+	rsJson := RunningStatsJSON{
+		RawVal:    rs.rawVal.CVal,
+		RangeStat: rs.rangeStat,
+		AvgStat:   rs.avgStat,
+	}
+	if rs.hll != nil {
+		rsJson.Hll = rs.hll.ToBytes()
+	}
+	if rs.rawVal.Dtype == utils.SS_DT_STRING_SET {
+		rsJson.StrSet = rs.rawVal.CVal.(map[string]struct{})
+		rsJson.RawVal = nil
+	}
+	if rs.rawVal.Dtype == utils.SS_DT_STRING_SLICE {
+		rsJson.StrList = rs.rawVal.CVal.([]string)
+		rsJson.RawVal = nil
+	}
+
+	return rsJson
+}
+
+func (rj RunningStatsJSON) GetRunningStats() (runningStats, error) {
+	rs := runningStats{
+		rangeStat: rj.RangeStat,
+		avgStat:   rj.AvgStat,
+	}
+	if rj.RawVal != nil {
+		CVal := utils.CValueEnclosure{}
+		err := CVal.ConvertValue(rj.RawVal)
+		if err != nil {
+			return runningStats{}, fmt.Errorf("RunningStatsJSON.GetRunningStats: failed to convert value, err: %v", err)
+		}
```

Comment:
nit: also log this error. You can use `utils.TeeErrorf()` to log and return an error

Same comment elsewhere

Reply:
addressed in the latest commit.